step_exec_lib/utils/processes.py

```diff
@@ -10,6 +10,29 @@ def run_and_log(args: List[str], **kwargs: Any) -> subprocess.CompletedProcess:
     logger.info(" ".join(args))
     if "text" not in kwargs:
         kwargs["text"] = True
+
     run_res = subprocess.run(args, **kwargs)  # nosec
     logger.info(f"Command executed, exit code: {run_res.returncode}.")
     return run_res
+
+
+def run_and_handle_error(args: List[str], error_text: str, **kwargs: Any) -> int:
+    logger.info("Running command:")
+    logger.info(" ".join(args))
+    if "text" not in kwargs:
+        kwargs["text"] = True
```

Comment:
Maybe I don't know enough about python but why is this needed?

Reply:
@MarcelMue This is the same as `run_and_log` just with the extra error handling needed for the go test executor.

@piontec Can you help with why we need this?

Reply:
It's part of how `run` works, see: https://docs.python.org/3/library/subprocess.html#subprocess.run

Basically it means "parse stdout as encoded text".